frontend/src/components/Admin/CoffeeChats/CoffeeChatDetails.tsx

```diff
@@ -114,6 +117,31 @@ const CoffeeChatDetails: React.FC = () => {
     return map;
   }, [coffeeChats]);
 
+  const runAutoCheckerForCategory = async (category: string) => {
```

Comment:
I would move this logic to the backend/API layer, since you're doing multiple data operations here. Take a look at how `regradeSubmissions` is implemented for Dev Portfolios for inspiration.